src/main/java/dataStructures/hashSet/openAddressing/HashSet.java

```diff
@@ -44,41 +45,44 @@ public HashSet() {
         this.buckets = tempBuckets;
         this.size = 0;
 
-        // There is no way to retrieve an instance of Tombstone. Therefore, it is safe to cast Tombstone to T.
-        @SuppressWarnings("unchecked")
-        T tempVar = (T) Tombstone.TOMBSTONE;
-        this.tombstone = tempVar;
     }
 
     /**
-     * Adds the specified element to this set if it is not already present
+     * Adds the specified element to this set if it is not already present.
      * If this set already contains the element, the call leaves the set unchanged and returns false.
      * <p>
      * If load factor (0.75) is exceeded, triggers a resize operation and double the current capacity.
      * It's important to note that resizing is not performed with every add operation but rather when the load
-     * factor exceeds the threshold. Therefore, the amortized time complexity of adding elements remains O(1)
+     * factor exceeds the threshold. Therefore, the amortized time complexity of adding elements remains O(1).
      *
-     * @param element the element to be added to this set
-     * @return true if this set did not already contain the specified
-     *     element
+     * @param element the element to be added to this set.
+     * @return true if this set did not already contain the specified element.
      */
     public boolean add(T element) {
-        if (this.contains(element)) {
-            return false;
+        if (contains(element)) {
+            return false; // Element is not added.
         }
 
         if (isLoadFactorExceeded()) {
-            resize(this.capacity() * 2); // Resize to double the capacity.
+            resize(capacity() * 2); // Resize to double the capacity.
         }
 
-        int bucketIndex = this.linearProbe(element);
-        if (!this.isEmptyBucket(
-            bucketIndex)) { // probe function returns the index of an empty bucket or the index containing the element.
-            return false; // Duplicate elements are not added to the set.
+        int collisions = 0;
+        while (collisions < capacity()) {
+            int bucketIndex = hashFunction(element, collisions);
+
+            // Insert into empty bucket.
+            if (isEmptyBucket(bucketIndex)) {
```

Comment:
minor issue: add in-line comment reminding empty is defined as either null or tombstone. Since u have 3 functions that at first glance, to a new reader, might seem the same

Reply:
Resolved in a1d46976c1168ff2b597453fa8d7876f3dc57d7b